src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String SEPARATOR = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String[] lines = readFile(fromFileName);
+            int totalSupply = 0;
+            int totalBuy = 0;
+
+            for (String line : lines) {
+                String[] fields = line.split(SEPARATOR);
+                String operation = fields[OPERATION_INDEX];
+                int amount = Integer.parseInt(fields[AMOUNT_INDEX]);
+                if (OPERATION_SUPPLY.equals(operation)) {
+                    totalSupply += amount;
+                } else if (OPERATION_BUY.equals(operation)) {
+                    totalBuy += amount;
+                }
+            }
+
+            int result = calculateResult(totalSupply, totalBuy);
+            String report = generateReport(totalSupply, totalBuy, result);
+            writeToFile(report, toFileName);
+        } catch (IOException e) {
```

Comment:
this catch block should be in readFile() method, there is no need to pass exception handler to parent one